src/components/SupportersDistribution.js

```diff
@@ -0,0 +1,162 @@
+      maxIncluded: DISTRIBUTION_ITEMS_MAX,
+    }
+  ).map((stake, index) => ({
+    item: {
+      entity: stake.index === -1 ? 'Others' : stakes[stake.index].entity,
+      amount: formatTokenAmount(
+        stakes[stake.index].amount,
+        stakeToken.decimals
+      ),
+    },
+    percentage: stake.percentage,
+  }))
+}
+
+const SupportersDistribution = React.memo(function SupportersDistribution({
+  stakes,
+  totalTokensStaked,
+}) {
+  const theme = useTheme()
+  const { stakeToken } = useAppState()
+
+  const transformedStakes = useMemo(() => {
+    if (!stakes) {
+      return null
+    }
+    return displayedStakes(stakes, totalTokensStaked, stakeToken)
+  }, [stakes, totalTokensStaked, stakeToken])
+
+  const colors = [theme.green, theme.red, theme.purple, theme.yellow]
+
+  return (
+    <Box heading="Supported by" padding={2 * GU}>
+      <div>
+        <Distribution
+          colors={colors}
+          items={transformedStakes}
+          renderFullLegendItem={({ color, item, index, percentage }) => {
+            return (
+              <div
+                css={`
+                  display: flex;
+                  align-items: center;
+                  width: 100%;
+                `}
+              >
+                <div
+                  css={`
+                    background: ${color};
+                    width: 8px;
+                    height: 8px;
+                    margin-right: ${0.5 * GU}px;
+                    border-radius: 50%;
+                  `}
+                />
+                <DistributionItem
+                  amount={item.amount}
+                  entity={item.entity}
+                  percentage={percentage}
+                  stakeToken={stakeToken}
+                />
+              </div>
+            )
+          }}
+        />
+        <div
+          css={`
+            margin-top: ${3 * GU}px;
+            display: flex;
+            justify-content: space-between;
+            ${textStyle('body4')}
+          `}
+        >
+          <span>Total </span>
+          <span>
+            {formatTokenAmount(totalTokensStaked, stakeToken.decimals)}
+            {` `}
+            {stakeToken.symbol}
+          </span>
+        </div>
+      </div>
+    </Box>
+  )
+})
+
+const DistributionItem = ({ amount, entity, percentage, stakeToken }) => {
+  const theme = useTheme()
+  const { account } = useWallet()
+  const isCurrentUser = addressesEqual(entity, account)
+  const { layoutName } = useLayout()
+  const compactMode = layoutName === 'medium' || layoutName === 'small'
+
+  return (
+    <div
+      css={`
+        display: flex;
+        width: 100%;
+        justify-content: space-between;
+      `}
+    >
+      <IdentityBadge
+        entity={entity}
+        connectedAccount={isCurrentUser}
+        compact
+        css={`
+          width: ${compactMode ? 'auto' : '110px'};
+        `}
+      />
+      <div
+        css={`
+          display: flex;
+          float: right;
```

Comment:
Don't think the `float` prop is needed here. We are also missing the center alignment.

We could also move the `textStyle` directly to this parent node.

```suggestion
    align-items: center;
    ${textStyle('body4')};
```

Reply:
why do you think that there is a need to align only the percentage and the honey value? it will be different from the label alignment

Reply:
![Screen Shot 2020-08-17 at 7 42 41 PM](https://user-images.githubusercontent.com/22663232/90451354-00ce0380-e0c2-11ea-9c62-b2bbcfb450a2.png)

The label `div` wrapping the `%` and `amount` is stretched due to the `label` being bigger.
Aligning the items to the center would also align them indirectly with the label.